.ci/windows-steps.yaml

```diff
@@ -0,0 +1,169 @@
+
+# Configure mlpack
+- powershell: |
+    git clone --depth 1 https://github.com/mlpack/mlpack.git
+    mkdir mlpack\build
+    cd mlpack\build
+
+    cmake $(CMakeGenerator) `
+    $(CMakeArgs-mlpack) `
+    -DBLAS_LIBRARIES:FILEPATH=$(Agent.ToolsDirectory)\OpenBLAS.0.2.14.1\lib\native\lib\x64\libopenblas.dll.a `
+    -DLAPACK_LIBRARIES:FILEPATH=$(Agent.ToolsDirectory)\OpenBLAS.0.2.14.1\lib\native\lib\x64\libopenblas.dll.a `
+    -DARMADILLO_INCLUDE_DIR="..\..\armadillo-8.400.0\include" `
+    -DARMADILLO_LIBRARY="..\..\armadillo-8.400.0\Release\armadillo.lib" `
+    -DBOOST_INCLUDEDIR=$(Agent.ToolsDirectory)\boost.1.60.0.0\lib\native\include `
+    -DBOOST_LIBRARYDIR=$(Agent.ToolsDirectory)\boost_libs `
+    -DBUILD_JULIA_BINDINGS=OFF `
+    -DCMAKE_BUILD_TYPE=Release ..
+  displayName: 'Configure mlpack'
+
+# Build mlpack
+- task: MSBuild@1
+  inputs:
+    solution: 'mlpack/build/*.sln'
+    msbuildLocationMethod: 'location'
+    msbuildVersion: $(MSBuildVersion)
+    configuration: 'Release'
+    msbuildArchitecture: 'x64'
+    msbuildArguments: /m /p:BuildInParallel=true
+    maximumCpuCount: false
+    clean: false
+  displayName: 'Build mlpack'
+
+# Configure mlpack
+- powershell: |
+    cp $(Agent.ToolsDirectory)\boost_libs\*.* mlpack\build\
+    cp $(Agent.ToolsDirectory)\OpenBLAS.0.2.14.1\lib\native\lib\x64\*.* mlpack\build\
+    cp $(Agent.ToolsDirectory)\OpenBLAS.0.2.14.1\lib\native\bin\x64\*.* mlpack\build\
+  displayName: 'Configure mlpack'
+
+# Configure protobuf
+- bash: |
+    git clone https://github.com/protocolbuffers/protobuf.git
+    cd protobuf
+    git submodule update --init --recursive
+    cd cmake
+    mkdir solution && cd solution
+    cmake $(CMakeGenerator)  -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=../../install ..
+  displayName: 'Configure protobuf'
+
+# Build protobuf
+- task: MSBuild@1
+  inputs:
+    solution: 'protobuf/cmake/solution/*.sln'
+    msbuildLocationMethod: 'location'
+    msbuildVersion: $(MSBuildVersion)
+    configuration: 'Release'
+    msbuildArchitecture: 'x64'
+    msbuildArguments: /m /p:BuildInParallel=true
+    clean: false
+    maximumCpuCount: false
+  displayName: 'Build protobuf'
+
+- powershell: |
+    mkdir build
+    cp $(Agent.ToolsDirectory)\boost_libs\*.* build\
+    cp $(Agent.ToolsDirectory)\OpenBLAS.0.2.14.1\lib\native\lib\x64\*.* build\
+    cp $(Agent.ToolsDirectory)\OpenBLAS.0.2.14.1\lib\native\bin\x64\*.* build\
+    cd build
+    cmake $(CMakeGenerator) `
+    -DBLAS_LIBRARIES:FILEPATH=$(Agent.ToolsDirectory)\OpenBLAS.0.2.14.1\lib\native\lib\x64\libopenblas.dll.a `
+    -DLAPACK_LIBRARIES:FILEPATH=$(Agent.ToolsDirectory)\OpenBLAS.0.2.14.1\lib\native\lib\x64\libopenblas.dll.a `
+    -DARMADILLO_INCLUDE_DIR="..\armadillo-8.400.0\include" `
+    -DARMADILLO_LIBRARY="..\armadillo-8.400.0\Release\armadillo.lib" `
+    -DProtobuf_INCLUDE_DIR="..\protobuf\src\" `
+    -DPROTOBUF_LIBRARY="..\protobuf\cmake\solution\Release\libprotobuf.lib" `
```

Comment:
I guess you already checked that `libprotobuf.lib` is in `..\protobuf\cmake\solution\Release\libprotobuf.lib` before the build?

Might be  good to copy `libprotobuf.*` into `build\` as well make sure it's there and use `$(Agent.ToolsDirectory)\boost_libs\libprotobuf.lib` instead.

Debugging a windows build is often not as easy.